application/ui/components/input/array.jsx

```diff
@@ -0,0 +1,147 @@
+/** @jsx React.DOM */
+'use strict';
+
+var _             = require('underscore');
+var React         = require('react');
+
+module.exports = React.createClass({
+
+    displayName : 'ArrayParameter',
+
+    getInitialState : function()
+    {
+        return {
+            values   : [],
+            numItems : 0
```

Comment:
no, as intermediate items could be removed...we'd need to renumber the array every time we did that.  this way we just keep pushing values and not have to worry about re-numerating it until we actually go to pass the value back up the stack